### DISCUSSION TITLE
Deep Selector - Can deep selector format be used for XPATH? Does deep selector support XPATH

### DISCUSSION THREAD

**Discussion by umer-ali-khan** [webdriverio/webdriverio#11522] [Q&A] [answered]
Hi

Our application is using shadow DOM and wdio examples recommend to use deep selectors. But in the examples deep selector use CSS locators like $('>>>some css');

Questions:
1. does deep selector support xpaths or only css can be used with deep selectors?
2. how deep selector be used for dynamic locators via CSS or XPATHs?
Please advise.

**Comment by christian-bromann** [comment]
> 1. does deep selector support xpaths or only css can be used with deep selectors?

Only css selectors for now. Feel free to make a contribution to https://github.com/webdriverio/query-selector-shadow-dom 😉 

> 2\. how deep selector be used for dynamic locators via CSS or XPATHs?

What are dynamic locators?

**Reply by umer-ali-khan** [reply]
Dynamic locators are when we substitute a value inside the common locator to find appropriate values. Say a common locator to select the dropdown value where only dropdown values needs to be replaced for each different option to click on.

**Reply by umer-ali-khan** [reply]
having said that does shadow$ or shadow$$ support xpaths? I presume they do but please confirm

**Reply by christian-bromann** [reply, marked as answer]
Yes, they do.
